MiniAn/minian_parameters.py

```diff
@@ -0,0 +1,335 @@
+
+        self.params_get_noise_fft = {
+            'noise_range': (noise_freq, 0.5)
+        }
+        if "get_noise_fft" in advanced_settings:
+            self.params_get_noise_fft, advanced_settings["get_noise_fft"] = self.update_func_advanced_param(self.params_get_noise_fft, advanced_settings["get_noise_fft"], mnCnmf.get_noise_fft)
+
+        self.params_update_spatial = {
+            'dl_wnd': neuron_diameter[-1],
+            'sparse_penal': parameters[mn_defs.ParametersKeys.SPATIAL_PENALTY],
+            'size_thres': (np.ceil(0.9*(np.pi*neuron_diameter[0]/2)**2), np.ceil(1.1*(np.pi*neuron_diameter[-1]/2)**2))
+        }
+        if "update_spatial" in advanced_settings:
+            self.params_update_spatial, advanced_settings["update_spatial"] = self.update_func_advanced_param(self.params_update_spatial, advanced_settings["update_spatial"], mnCnmf.update_spatial)
+
+        self.params_update_temporal = {
+            'noise_freq': noise_freq,
+            'sparse_penal': parameters[mn_defs.ParametersKeys.TEMPORAL_PENALTY],
+            'p': 1,
+            'add_lag': 20,
+            'jac_thres': 0.2
+        }
+        if "update_temporal" in advanced_settings:
+            self.params_update_temporal, advanced_settings["update_temporal"] = self.update_func_advanced_param(self.params_update_temporal, advanced_settings["update_temporal"], mnCnmf.update_temporal)
+
+        # Update AdvancedSettings for self.parameters
+        self.parameters[mn_defs.ParametersKeys.ADVANCED_SETTINGS] = advanced_settings.copy()
+
+
+    #--------------------------------------------- functions for advanced parameters -------------------------------------------------------------------------
+    def remove_keys_not_in_fun_arguments(self,
+        input_dic:dict, func
+        ) -> dict:
+        '''
+        This function while keep the keys in input_dic that are not use in the function func
+
+        Returns
+        -------
+        The new dictionary new_dictionary
+        '''
+        func_arguments = inspect.getfullargspec(func).args
+        new_dictionary = {key: input_dic[key] for key in input_dic if key in func_arguments}
+        return new_dictionary
+
+    def update_func_advanced_param(self,
+        param_func,
+        advanced_parameters,
+        func
+        ):
+        '''
+        This function while change the value of the key from the dictionary param with the value of the key from the dictionary advanced_parameters.
+        It while also remove the keys from the dictionary advanced_parameters that are not used in the function func
+
+        Returns
+        -------
+        it while return the dictionary param_func with the new values and also the new advanced_parameters with only the used keys
+
+        '''
+        advanced_parameters = self.remove_keys_not_in_fun_arguments(advanced_parameters, func)
```

Comment:
Does it make sense to have a separate function for that?

Reply:
No I removed it